components/nimbus/src/tests/test_enrollment.rs

```diff
@@ -3136,3 +3138,61 @@ fn test_populate_feature_maps() -> Result<()> {
 
     Ok(())
 }
+
+#[test]
+fn test_sort_experiments_by_published_date() -> Result<()> {
+    let slug_1 = "slug-1";
+    let slug_2 = "slug-2";
+    let slug_3 = "slug-3";
+    let slug_4 = "slug-4";
+    let experiments = vec![
+        get_experiment_with_published_date(slug_1, None),
+        get_experiment_with_published_date(slug_2, Some("2023-11-21T18:00:00Z".into())),
+        get_experiment_with_published_date(slug_3, None),
+        get_experiment_with_published_date(slug_4, Some("2023-11-21T15:00:00Z".into())),
+    ];
+    let result = sort_experiments_by_published_date(&experiments);
+
+    assert_eq!(slug_1, result[0].slug);
+    assert_eq!(slug_3, result[1].slug);
+    assert_eq!(slug_4, result[2].slug);
+    assert_eq!(slug_2, result[3].slug);
+    Ok(())
+}
+
+#[test]
+fn test_evolve_enrollments_ordering() -> Result<()> {
+    let _ = env_logger::try_init();
+    let (_, app_ctx, aru) = local_ctx();
+    let th = app_ctx.into();
+    let ids = HashSet::new();
+    let evolver = EnrollmentsEvolver::new(&aru, &th, &ids);
+
+    let exp1 = get_single_feature_experiment_with(
+        "exp1",
+        "colliding",
+        json!({"x": 1 }),
+        json!({"publishedDate": "2023-11-21T18:00:00Z"}),
+    );
+    let exp2 = get_single_feature_experiment_with(
+        "exp2",
+        "colliding",
+        json!({"x": 2 }),
+        json!({"publishedDate": "2023-11-21T15:00:00Z"}),
+    );
```

Comment:
Two things: could you change the argument values to be a little more self documenting:

```suggestion
    let exp1 = get_single_feature_experiment_with(
        "slug_1",
        "colliding-feature",
        json!({"x": 1 }),
        json!({"publishedDate": "2023-11-21T18:00:00Z"}),
    );
    let exp2 = get_single_feature_experiment_with(
        "slug_2",
        "colliding-feature",
        json!({"x": 2 }),
        json!({"publishedDate": "2023-11-21T15:00:00Z"}),
    );
```

Second: I wonder if this could `get_single_feature_experiment_with` might be more clear as:

```rs
impl Experiment {
   fn patch(self, patch: &Value) -> Self {
       // …
   }
}
```

which you can then use:

```rs
let exp1 = get_single_feature_experiment(…).patch(json!({"publishedDate": ""}))
```